src/python/pants/bin/daemon_pants_runner.py

```diff
@@ -304,6 +306,7 @@ def run(self):
         # excepthook catches `SignalHandledNonLocalExit`s from signal handlers, which isn't
         # happening here, so something is probably overriding the excepthook. By catching Exception
         # and calling this method, we emulate the normal, expected sys.excepthook override.
-        ExceptionSink._log_unhandled_exception_and_exit(exc=e)
+        ExceptionSink._log_unhandled_exception(exc=e)
```

Comment:
Would calling `ExceptionSink.reset_exiter(self._exiter)` here above this line also resolve the issue described?

Reply:
It wouldn't because we can't use the global exiter.

This is because `PantsDaemon` might be on another thread and wanting to run it's own exiter, so we might end up on a race condition of "who has overwritten the global exiter first"

Reply:
Would it be reasonable to consider handling that case with some thread-aware logic in the `Exiter` instead of having to introduce complexity into invoking each runner? Which exiter is intended to prevail in that case?

Reply:
After some experimentation, I'm pretty sure I agree with you and this is not the best way to handle this situation. Also, the multithreaded nature of the code does not affect exiters, because they will do what we want even in the presence of race conditions (i.e. all the exceptions will be logged somewhere).